### DISCUSSION TITLE
How to use with @vanilla-extract/esbuild-plugin

### DISCUSSION THREAD

**Discussion by wadamek65** [egoist/tsup#1191] [Q&A]
I'm trying to get this to work but unfortunately I run into issues. Just setting up the plugin like instructed by the docs doesn't work:

```ts
import { defineConfig } from 'tsup';
import { vanillaExtractPlugin } from '@vanilla-extract/esbuild-plugin';

export default defineConfig({
  format: ['esm'],
  watch: true,
  esbuildPlugins: [vanillaExtractPlugin()]
});
```

Neither does help setting/unsetting the `bundle` flag. No CSS files are generated and the `.css.ts` files are included in the compiled files which is obviously throwing errors when imported. Any help or pointers? [Link to vanilla-extract docs](https://vanilla-extract.style/documentation/integrations/esbuild/)

**Comment by lbdavis1234** [comment]
It sounds like you may not have the base vanilla extract css package installed. 

This is the setup that I have working:

1. Install `@vanilla-extract/css` and `@vanilla-extract/esbuild-plugin` as dev dependencies.
2. In your `package.json`, add an entry for "style" similar to what you would do for types, like:  `style": "./dist/index.css`.
3. Run the build command -- it should now output your vanilla-extract css in `dist/index.css`.
4. At the top level of the app where you're using the css, import like `import "my-app/dist/index.css";`. 


Full example `package.json` for the project you're trying to build with tsup and vanilla extract
```json
{
  "name": "my-app",
  "version": "0.0.0",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./src/index.tsx",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    },
  },
  "main": "./dist/index.js",
  "module": "./dist/index.esm.js",
  "types": "./dist/index.d.ts",
  "style": "./dist/index.css", //<- Add path to outputs
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch",
  },
  "dependencies": {
    "react": "^18.2.0"
  },
  "devDependencies": {
    "@types/react": "^18.2.61",
    "@types/react-dom": "^18.2.19",
    "@vanilla-extract/css": "^1.17.4", //<- Add base package
    "@vanilla-extract/esbuild-plugin": "^2.3.18", //<- Add esbuild plugin package
    "tsup": "^8.0.2",
    "typescript": "5.5.4"
  }
}
```


Full example `tsup.config.ts`
```ts
import { defineConfig } from "tsup";
import { vanillaExtractPlugin } from "@vanilla-extract/esbuild-plugin";

export default defineConfig((options) => ({
  entryPoints: ["src/index.tsx"],
  format: ["cjs"],
  watch: true,
  esbuildPlugins: [vanillaExtractPlugin()], //<- Add vanillaExtractPlugin
  ...options,
}));
```
